### ISSUE TITLE
Control term is not accounting for time delta

### ISSUE THREAD

**Issue by improve100** [cra-ros-pkg/robot_localization#690]
TM: editing issue for clarity.

`prepareControl` doesn't appear to be using the `predictionDelta` term to compute an acceleration from the error between the controlled velocity and the current velocity.

https://github.com/cra-ros-pkg/robot_localization/blob/da0f9676c27db71b9e7cc76fbb227c92202ffd49/src/filter_base.cpp#L351-L375

Should there be a `controlAcceleration_(controlInd) /= predictionDelta;` line after line 371?

**Comment by ayrton04** [comment]
Yes, I think you're right. The [original code](https://github.com/cra-ros-pkg/robot_localization/commit/6e83ef2f3635fc84efc2413db0fc05e294c4d07e#diff-b96ffbfcfc303ae826323830be2c943272c096bf8cb93fdb56fef4c949a19ce2R283) was multiplying the error term by a frequency, and that frequency was computed as `1.0 / predictionDelta`:

https://github.com/cra-ros-pkg/robot_localization/commit/6e83ef2f3635fc84efc2413db0fc05e294c4d07e#diff-8a83a6a31c3794ee91112354ac3dcceaa70fb760089c851e5998dcc186dbaf65R329
https://github.com/cra-ros-pkg/robot_localization/commit/6e83ef2f3635fc84efc2413db0fc05e294c4d07e#diff-b96ffbfcfc303ae826323830be2c943272c096bf8cb93fdb56fef4c949a19ce2R283

But then we capped the acceleration with a gain term instead, and lost the time delta term. However, I don't think your proposed fix will work either. The acceleration limits get applied in `computeControlAcceleration`, so we need to pass the `predictionDelta` into that method so we can compute the final acceleration value within it, and then cap it. In other words, I think [this line](https://github.com/cra-ros-pkg/robot_localization/blob/da0f9676c27db71b9e7cc76fbb227c92202ffd49/include/robot_localization/filter_base.h#L360) needs to be:

`const double finalAccel = std::min(std::max(gain * error / predictionDelta, -limit), limit);`

But in order to do that, you need to pass `predictionDelta` into `computeControlAcceleration`, preferably as the third argument.

**Comment by ayrton04** [comment]
Care to submit a PR?
